File: Frontend/write.php
<?php
session_start();
require 'vendor/autoload.php';
use PhpAmqpLib\Connection\AMQPStreamConnection;
use PhpAmqpLib\Message\AMQPMessage;

if (isset($_SESSION['email']) && isset($_SESSION['password']) && isset($_SESSION['first']) && isset($_SESSION['last']) && isset($_SESSION['height']) && isset($_SESSION['weight']) && isset($_SESSION['movie']) && isset($_SESSION['color'])) {
    $email = $_SESSION['email'];
    $password = $_SESSION['password'];
    $first = $_SESSION['first'];
    $last = $_SESSION['last'];
    $height = $_SESSION['height'];
    $weight = $_SESSION['weight'];
    $color = $_SESSION['color'];
    $movie = $_SESSION['movie'];

    $rabbitMQHosts = '10.147.17.79';
    $rabbitMQPort = 5672;
    $rabbitMQUser = 'backend';
    $rabbitMQPassword = 'password';
    $virtualHost = '/';
    $exchangeName = 'frontend-backend';

    echo "name: " .$first . " " . $last . " email: " . $email . " pass: " . $password . " height: " . $height . " weight: " . $weight;

    try {
        // Create a connection to RabbitMQ
        $connection = new AMQPStreamConnection($rabbitMQHosts, $rabbitMQPort, $rabbitMQUser, $rabbitMQPassword, $virtualHost);

        // Create a channel
        $channel = $connection->channel();

        // Declare the exchange (direct type)
        $channel->exchange_declare($exchangeName, 'direct', false, true, false);

        // Create a message
        $messageBody = json_encode([
            'first_name' => $first,
            'last_name' => $last,
            'email' => $email,
            'password' => $password,
            'height' => $height,
            'weight' => $weight,
            'movie' => $movie,
            'color' => $color,
            'goal' => 'null',
        ]);

        // Specify the routing key for the backend queue
        $routingKey = 'reg.request'; // frontend

        // Publish the message to the exchange with the routing key
        $message = new AMQPMessage($messageBody);
        $channel->basic_publish($message, $exchangeName, $routingKey);

        // Close the channel and the connection
        $channel->close();
        $connection->close();

        // Clear the session variables
        unset($_SESSION['email']);
        unset($_SESSION['password']);
        unset($_SESSION['first']);
        unset($_SESSION['last']);
        unset($_SESSION['height']);
        unset($_SESSION['weight']);

        echo "Message sent to the exchange with routing key '$routingKey': $messageBody\n";
    } catch (\Exception $e) {
        echo "Error: " . $e->getMessage() . "\n";
    }
} else {
    echo "Registration did not succeed.";
}
?>

File: Database/Working Files/write.py
import pika

# RabbitMQ server connection parameters
rabbitmq_host = '10.248.179.22'
rabbitmq_port = 5672  # Default RabbitMQ port
rabbitmq_user = 'database'
rabbitmq_password = 'password'
virtual_host = '/'

# Exchange name (frontend-backend-direct-exchange)
exchange_name = 'backend-database'  # Replace with the actual exchange name

# Routing key for the frontend queue
routing_key = 'backend'

# Message content
message_body = 'Hello, backend!'

try:
    # Create a connection to RabbitMQ
    credentials = pika.PlainCredentials(rabbitmq_user, rabbitmq_password)
    parameters = pika.ConnectionParameters(rabbitmq_host, rabbitmq_port, virtual_host, credentials)
    connection = pika.BlockingConnection(parameters)

    # Create a channel
    channel = connection.channel()

    # Declare the exchange (direct type)
    channel.exchange_declare(exchange=exchange_name, exchange_type='direct', durable=True)

    # Publish the message with the routing key
    channel.basic_publish(
        exchange=exchange_name,
        routing_key=routing_key,
        body=message_body,
        properties=pika.BasicProperties(
            delivery_mode=2,  # Make the message persistent
        )
    )

    print(f"Message sent to the exchange '{exchange_name}' with routing key '{routing_key}': {message_body}")

    # Close the channel and the connection
    connection.close()

except Exception as e:
    print(f"Error: {e}")

